### PR TITLE
Publish the cl_img_swap_ops extension specification.

### PR DESCRIPTION
**Pull request by tomasz-platek**
Publish the cl_img_swap_ops extension specification.

### CODE REVIEW

File: extensions/cl_img_swap_ops.asciidoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Copyright (c) 2024 Imagination Technologies Ltd. All Rights Reserved.
+
+== Status
+
+Final Draft
+
+== Version
+
+Built On: {docdate} +
+Version: Major.Minor.Patch
```
Comment (bashbaug) [inline extensions/cl_img_swap_ops.asciidoc:35]:
```suggestion
Version: 1.0.0
```

File: extensions/cl_img_swap_ops.asciidoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+== Modifications to the OpenCL C Specification
+
+(Add to Table 16 - Built-in Scalar and Vector Argument Common Functions in Section 6.15.4 - Common Functions) ::
++
+--
+[cols="1,2",options="header"]
+|====
+| Function | Description
+| gentype *img_swap_x*(gentype value)
+    a| `img_swap_x` swaps `values` between work-items in the following way: all work-items are divided into blocks of four consecutive elements. For each block:
```
Comment (bashbaug) [inline extensions/cl_img_swap_ops.asciidoc:71]: Consider documenting the behavior if the number of work-items is not evenly divisible by four, or if some work-items in the block of four are inactive (due to flow control).

Also consider documenting what "consecutive" means, especially for multi-dimensional work-groups.

**Review by bashbaug — APPROVED** [review]

File: extensions/cl_img_swap_ops.asciidoc (hunk 2 above)
Comment (tomasz-platek) [reply extensions/cl_img_swap_ops.asciidoc:71]: I defined the behavior as undefined for these cases as I don't see any reasonable way of handling them.

**Comment by bashbaug** [comment]
PR has been open for over a month with no additional comments.  Merging!

**Pull request merged**